src/controllers/features/common/checkSavedApplicationController.ts

```diff
@@ -1,30 +1,60 @@
 import { NextFunction, Request, Response } from "express";
 import { selectLang, addLangToUrl, getLocalesService } from "../../../utils/localise";
-import { BASE_URL, CANNOT_SUBMIT_ANOTHER_APPLICATION, CHECK_SAVED_APPLICATION, SAVED_APPLICATION, TYPE_OF_BUSINESS } from "../../../types/pageURL";
+import { BASE_URL, CANNOT_REGISTER_AGAIN, CANNOT_SUBMIT_ANOTHER_APPLICATION, CHECK_SAVED_APPLICATION, SAVED_APPLICATION, TYPE_OF_BUSINESS } from "../../../types/pageURL";
 import { Session } from "@companieshouse/node-session-handler";
-import { getSavedApplication } from "../../../services/acspRegistrationService";
 import logger from "../../../utils/logger";
-import { saveDataInSession } from "../../../common/__utils/sessionHelper";
 import { ErrorService } from "../../../services/errorService";
+import { Resource } from "@companieshouse/api-sdk-node";
+import { ApiErrorResponse } from "@companieshouse/api-sdk-node/dist/services/resource";
+import { TransactionList } from "@companieshouse/api-sdk-node/dist/services/transaction/types";
+import { getSavedApplication } from "../../../services/transactions/transaction_service";
+import { deleteAcspApplication } from "../../../services/acspRegistrationService";
 
 export const get = async (req: Request, res: Response, next: NextFunction) => {
     const session: Session = req.session as any as Session;
-    const savedApplication = await getSavedApplication(session, res.locals.userId);
+    const savedApplication : Resource<TransactionList> | ApiErrorResponse = await getSavedApplication(session, res.locals.userId);
+    logger.debug("savedApplication: " + savedApplication);
     const lang = selectLang(req.query.lang);
     const locales = getLocalesService();
-    if (savedApplication.status === 404) {
-        // Direct to start a new journey if no application found or application was rejected
-        saveDataInSession(req, "resume_application", false);
+
+    if (savedApplication.httpStatusCode === 200) {
+        logger.debug("saved application exists");
+        const transactionlistResource = savedApplication as Resource<TransactionList>;
+        const transactionList = transactionlistResource.resource;
+        const redirectionUrl = getRedirectionUrl(transactionList, session, res, locales, lang);
```

Comment:
You can pass the Full transaction Resource to the new function. And also move this function to a service layer please.

Reply:
resolved